service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedDataRepoSnapshotResource.java

```diff
@@ -103,6 +106,12 @@ public void validate() {
     }
   }
 
+  @Override
+  public boolean checkAccess(FlightBeanBag context, AuthenticatedUserRequest userReq) {
+    DataRepoService dataRepoService = context.getDataRepoService();
+    return dataRepoService.snapshotExists(instanceName, snapshotId, userReq);
```

Comment:
It looks like this is doing something different than `checkAccess` for other resource types (exists vs canRead). Would it make sense for this to return false if the snapshot is not found or not accessible by the user, similar to the other resource types?

Note that the CLI does not currently support TDR snapshots, so this won't affect the usage of this endpoint right now.

Reply:
This ends up calling the `retrieveSnapshot` on TDR which checks for READ_DATA access, so as I understand this actually is checking for read access and not just existence. Renamed the method in DataRepoService for clarity